### PR TITLE
refactor: add page and limit to urlParams 

### LINKED ISSUE #712: feat: Implement support for page meta query params in page Url
> Can we add the page number to the URL?
> `https://deploy-preview-708--oss-insights.netlify.app/javascript/repositories/filter/top-100-repos?page=4`

This is not present on main, but would make it easier for the user to navigate

Sure, this is a better approach to pagination in general. we could also move other page meta queries to the URL query params as well. Example: `https://deploy-preview-708--oss-insights.netlify.app/javascript/repositories/filter/top-100-repos?page=4&limit=10`

would suggest we create a follow-up issue for this.

_Originally posted by @SunGoldTech in https://github.com/open-sauced/insights/issues/708#issuecomment-1362887074_

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR 

- Adds page and limit query to URLParams accross the application
- refactors string to number conversion also for the routers queryParams

### TODO

- [x] Add page and limit query to insights and list index page
- [x] Add page query to insights pages
- [x] Add page query to list dashboard pages
- [x] add page query to explore page

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #712

### Insights page

https://github.com/open-sauced/app/assets/62995161/d196eec2-5f57-4cc2-9d0f-a0ccda1b304f

### Highlights Page

https://github.com/open-sauced/app/assets/62995161/aeccff14-376a-4977-9048-760bdf24cd96

### Explore page

https://github.com/open-sauced/app/assets/62995161/9d833bc0-8ac2-4ef0-a861-0aa27d7e8935

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
We have a duplicate story with same id that breaks this build
image

One in the molecules dir and the other in graphs
https://github.com/OgDev-01/app-opensauced/blob/39442930dcc13773cd9e3fb5702060c25816c823/components/Graphs/ContributionsTreemap/contributions-treemap.stories.tsx and https://github.com/OgDev-01/app-opensauced/blob/f8092ee9cb9dde84cddc1c340e63fc7f2365a9f1/components/molecules/ContributionsTreemap/contributions-treemap.stories.tsx

The story mockup seems different, and I'm wondering if these two stories need to co-exist. If that is the case, we need to update the naming to be unique to avoid duplicate id's. cc: @nickytonline

**Comment by brandonroberts** [comment]
@OgDev-01 the ContributionsTreemap under `components/molecules` is gone, so there should only be one. We landed the API v2 change on beta and cleaned that up

**Comment by brandonroberts** [comment]
@OgDev-01 will you add a screen capture demo of the feature and fix the merge conflicts?

**Comment by nickytonline** [comment]
> We have a duplicate story with same id that breaks this build image
> 
> One in the molecules dir and the other in graphs [OgDev-01/app-opensauced@`3944293`/components/Graphs/ContributionsTreemap/contributions-treemap.stories.tsx](https://github.com/OgDev-01/app-opensauced/blob/39442930dcc13773cd9e3fb5702060c25816c823/components/Graphs/ContributionsTreemap/contributions-treemap.stories.tsx) and [OgDev-01/app-opensauced@`f8092ee`/components/molecules/ContributionsTreemap/contributions-treemap.stories.tsx](https://github.com/OgDev-01/app-opensauced/blob/f8092ee9cb9dde84cddc1c340e63fc7f2365a9f1/components/molecules/ContributionsTreemap/contributions-treemap.stories.tsx)
> 
> The story mockup seems different, and I'm wondering if these two stories need to co-exist. If that is the case, we need to update the naming to be unique to avoid duplicate id's. cc: @nickytonline

You'll need to get latest as there are no duplicate stories. It got moved in a PR a while back. You can see in beta.

**Comment by OgDev-01** [comment]
> @OgDev-01 will you add a screen capture demo of the feature and fix the merge conflicts?

On it now..

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 1]
```diff
@@ -393,7 +393,7 @@ const ContributorProfileHeader = ({
 const AddToListDropdown = ({ username }: { username: string }) => {
   const [selectListOpen, setSelectListOpen] = useState(false);
   const [selectedList, setSelectedList] = useState<OptionKeys[]>([]);
-  const { data } = useFetchAllLists();
+  const { data } = useFetchAllLists({});
```
Comment (nickytonline) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:396]: refactor (blocking): passing an empty object doesn't make sense, it all the properies are optional, you can default to an empty object in the hook.

**Review by nickytonline — CHANGES_REQUESTED** [review]
I'm hesitant to merge all these changes all at once. Can we break this PR into several so that we can do it either a couple pages at a time  or per hook?

**Comment by OgDev-01** [comment]
> I'm hesitant to merge all these changes all at once. Can we break this PR into several so that we can do it either a couple pages at a time or per hook?

Alright @nickytonline, I'll be breaking this into smaller pieces in several PRs during the weekend, if that is okay by you?

**Pull request closed without merging**